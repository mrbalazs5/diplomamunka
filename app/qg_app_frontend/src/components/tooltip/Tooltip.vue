<template>
  <div class="tooltip">
    <slot />

    <div class="helper">
      <span class="tooltiptext">{{ text }}</span>
    </div>
  </div>
</template>

<script>
export default {
  name: "Tooltip",
  props: {
    text: String
  }
}
</script>

<style lang="scss" scoped>
  .tooltip {
    position: relative;
    display: flex;

    .helper {
      position: absolute;
    }

    .tooltiptext {
      visibility: hidden;
      transition: 0.5s;
      opacity: 0;
      background-color: #5e5e5e;
      color: #fff;
      text-align: center;
      border-radius: 6px;
      padding: 5px 10px;
      margin-left: 27px;
      margin-top: 20px;
      font-size: 14px;
      position: fixed;
      z-index: 1;
      white-space: nowrap;
      pointer-events: none;
    }
  }

  .tooltip:hover .tooltiptext {
    visibility: visible;
    opacity: 1;
  }
</style>