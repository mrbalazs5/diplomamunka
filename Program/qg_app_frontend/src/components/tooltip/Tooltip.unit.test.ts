import { shallowMount } from "@vue/test-utils";
import { expect, test } from "vitest";
import Tooltip from "@/components/tooltip/Tooltip.vue";

test("mount component", () => {
    const mockText = "test";

    const wrapper = shallowMount(Tooltip, {
        props: {
            text: mockText,
        },
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.text()).toContain(mockText);
});
