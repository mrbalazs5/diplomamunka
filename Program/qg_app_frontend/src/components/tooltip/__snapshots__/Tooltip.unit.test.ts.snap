// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `
"<div class=\\"tooltip\\" data-v-213cff12=\\"\\">
  <div class=\\"helper\\" data-v-213cff12=\\"\\"><span class=\\"tooltiptext\\" data-v-213cff12=\\"\\">test</span></div>
</div>"
`;
